From cea125b12934c0d69b358b9405823a2fab2aeb08 Mon Sep 17 00:00:00 2001
From: Eirik Aavitsland <eirik.aavitsland@qt.io>
Date: Thu, 2 Oct 2025 09:28:57 +0200
Subject: [PATCH 109/239] Revert "QML generator: optimize by sharing custom
 easing curve objects"

This reverts commit 34cfd99f856d7ad6960e7a92eaea09c0c44476d9.

Can not work in 6.10, it depends on the easingCurve value type that is
only introduced in 6.11.

Change-Id: Ib98dea88fcbfed0fc4b7c12e11411590e9250c62
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
---
 .../generator/qquickqmlgenerator.cpp              | 15 +++------------
 .../generator/qquickqmlgenerator_p.h              |  2 --
 2 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/src/quickvectorimage/generator/qquickqmlgenerator.cpp b/src/quickvectorimage/generator/qquickqmlgenerator.cpp
index 97fca2820a..411d1a4f5d 100644
--- a/src/quickvectorimage/generator/qquickqmlgenerator.cpp
+++ b/src/quickvectorimage/generator/qquickqmlgenerator.cpp
@@ -344,11 +344,9 @@ void QQuickQmlGenerator::generateEasing(const QQuickAnimatedProperty::PropertyAn
 
         bool isLinear = (c1 == c1.transposed() && c2 == c2.transposed());
         if (!isLinear) {
-            int nextIdx = m_easings.size();
-            QString &id = m_easings[{c1.x(), c1.y(), c2.x(), c2.y()}];
-            if (id.isNull())
-                id = QString(QLatin1String("easing_%1")).arg(nextIdx, 2, 10, QLatin1Char('0'));
-            stream() << "easing: " << m_topLevelIdString << "." << id;
+            stream() << "easing.type: Easing.BezierSpline";
+            stream() << "easing.bezierCurve: [ " << c1.x() << ", " << c1.y()
+                     << ", " << c2.x() << ", " << c2.y() << ", 1, 1 ]";
         }
     }
 }
@@ -1183,13 +1181,6 @@ bool QQuickQmlGenerator::generateRootNode(const StructureNodeInfo &info)
             stream() << "}";
         }
 
-        for (const auto [coords, id] : m_easings.asKeyValueRange()) {
-            stream() << "property easingCurve " << id << ": { type: Easing.BezierSpline; bezierCurve: [ ";
-            for (auto coord : coords)
-                stream(SameLine) << coord << ", ";
-            stream(SameLine) << "1, 1 ] }";
-        }
-
         m_indentLevel--;
         stream() << "}";
         stream().flush();
diff --git a/src/quickvectorimage/generator/qquickqmlgenerator_p.h b/src/quickvectorimage/generator/qquickqmlgenerator_p.h
index dbe75686ce..f2883ab24f 100644
--- a/src/quickvectorimage/generator/qquickqmlgenerator_p.h
+++ b/src/quickvectorimage/generator/qquickqmlgenerator_p.h
@@ -21,7 +21,6 @@
 
 #include <QtCore/qtextstream.h>
 #include <QtCore/qbuffer.h>
-#include <QtCore/qmap.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -151,7 +150,6 @@ private:
     QString m_urlPrefix;
     QString m_topLevelIdString;
     QStringList m_extraImports;
-    QMap<std::array<qreal, 4>, QString> m_easings;
 };
 
 QT_END_NAMESPACE
-- 
2.51.2

